modules/core/exported/packet.go

```diff
@@ -13,24 +13,36 @@ type PacketI interface {
 	ValidateBasic() error
 }
 
-// Acknowledgement defines the interface used to return acknowledgements in the OnRecvPacket callback.
-// The Acknowledgement interface is used by core IBC to ensure partial state changes are not committed
+// RecvPacketStatus defines an enum type to signal the result status of an IBC application packet datagram and inform
+// core IBC of how to handle the application state changes held in the cache store which it was provided for OnRecvPacket callback execution.
+type RecvPacketStatus uint32
+
+const (
+	// Success instructs that the IBC application state should be persisted.
+	Success RecvPacketStatus = iota
+	// Failure instructs that the IBC application state should be discarded.
+	Failure
+	// Async instructs that the IBC application state should be persisted
+	// and acknowledgement data will be written later asynchronously.
+	Async
```

Comment:
Note for follow up, we might want these as `RecvStatusSuccess`, `RecvStatusFailure` or something like that..